### PR TITLE
Mh search endpoint

### PR DESCRIPTION
**Pull request by mikez321**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling (Styling to new or existing files)
- [ ] Testing
### Detailed Description
Adds a search endpoint to production
### Why is this change required? What problem does it solve?
This will allow FE to search for plants in our DB
### Was any previously written code changed?
Nope, just new stuff
### Directory modified:
Added test for endpoint and route for endpoint

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,34 @@
...
+__Get plant by ID__
+```
+oh-crop-be.herokuapp.com/api/v1/plants/2
+```
+
+Note: add a plant ID to the end... in the above example, plant ID = 2
+
+__Plant Search__
+```
+oh-crop-be.herokuapp.com/api/v1/plants/search?q=tomato
```
Comment (kelshadarby) [inline README.md:27]: This looks great, we should add some description on these as to what each endpoint does

File: app/__init__.py [hunk 2]
```diff
@@ -15,7 +15,6 @@ def create_app(config_name):
     app = FlaskAPI(__name__, instance_relative_config=True)
     app.config.from_object(app_config[config_name])
     app.config.from_pyfile('config.py')
-    # app.config['SQLALCHEMY_DATABASE_URI'] = os.environ['DATABASE_URL']
```
Comment (kelshadarby) [inline app/__init__.py:18]: We could probably remove this line if we aren't using

File: app/api/routes.py [hunk 3]
```diff
@@ -38,20 +43,30 @@ def all_plants():
...
+                'plant_image': plant.image
                 }
     response = jsonify(result)
     response.status_code = 200
     return response
+
+@api.route('/plants/search')
+def search_plants():
+    search = request.args['q']
+    plants = db.session.query(Plant).filter(Plant.plant_type.ilike('%{}%'.format(search))).all()
```
Comment (kelshadarby) [inline app/api/routes.py:60]: Nice work figuring this out!

File: run.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import os
 
 from app import create_app
+from dotenv import load_dotenv
```
Comment (kelshadarby) [inline run.py:4]: Added to ensure environment is running properly

File: tests/test_endpoint.py [hunk 5]
```diff
@@ -33,6 +33,20 @@ def test_api_can_return_a_plant_by_id(self):
...
+        jimmy = Plant(name='Jimothy',plant_type='Cherry Tomato',image='jim_photo.jpg',lighting='Full Sun',water_frequency=3,harvest_time=50,root_depth=12,annual=False)
+        agatha = Plant(name='Agatha',plant_type='Roma Tomato',image='agatha_photo.jpg',lighting='Full Sun',water_frequency=2,harvest_time=60,root_depth=12,annual=False)
+        dan = Plant(name='Dan',plant_type='Cactus',image='cactus_dan.jpg',lighting='Full Sun',water_frequency=7,harvest_time=None,root_depth=8,annual=False)
+        db.session.add_all([jimmy, agatha, dan])
+        db.session.commit()
+        res = self.client().get('/api/v1/plants/search?q=tomato')
+        self.assertEqual(res.status_code, 200)
+        self.assertNotIn(dan.plant_type, str(res.data))
+        self.assertIn(jimmy.plant_type, str(res.data))
+        self.assertIn(agatha.plant_type, str(res.data))
```
Comment (kelshadarby) [inline tests/test_endpoint.py:49]: We should add some edge cases late on
